Documentation/docs/learn/faq.md

```diff
@@ -0,0 +1,611 @@
+
+When the CMake source changes, you may see multiple prompts to reload
+many projects.
+
+DO NOT USE `Rebuild All` or `Batch Build`. Use the following recommended
+procedure.
+
+There is an ALL_BUILD target available in the MSVC++ workspace. The best
+way to build ITK is to set `ALL_BUILD` as the active project, then select
+the active configuration you want (`DEBUG`, etc.).
+
+This can also be done from the command line like this:
+
+```{bash}
+msdev ITK.dsw /MAKE "ALL_BUILD - Debug" /REBUILD
+```
+
+## Contribute
+
+### How do I make code contributions to ITK?
+
+Please see the [Contribute to ITK page](https://docs.itk.org/en/latest/contributing/index.html).
+
+## Testing
+
+### Is ITK tested?
+
+Please see this page: [ITK/Testing](https://itk.org/Wiki/ITK/Testing).
+
+## Working with image data
+
+### How do I iterate through all the pixels in an image?
+
+Please see the [`ImageRegionIterator.cxx`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Examples/Iterators/ImageRegionIterator.cxx)
+example.
+
+### What are Hounsfield Units?
+
+Hounsfield units were defined by Sir Godfrey N. Hounsfield, one of the
+pioneers of Computer Tomography for clinical applications.
+
+The units represent the linear attenuation of X-Rays for a particular
+material.
+
+The units scale is defined in such a way that level for Water is 0 and
+the level for Air is -1000.
+
+The attenuation of any other material is mapped linearly into this
+range.
+
+A table of Hounsfield units for common human tissues is avalable in the
+following link [ITK Hounsfield Units](https://itk.org/Wiki/ITK_Hounsfield_Units)
+
+### What are MRI T1 / T2 units?
+
+MRI images measure two types of relaxation times.
+
+The units are millisecons and the two relaxation concepts relate to how
+long it takes for the molecules in the tissue to realign themselves with
+a magentic field after being perturbed.
+
+Details on the MRI units are available on the [Relaxation (NMR)](https://en.wikipedia.org/wiki/Relaxation_%28NMR%29)
+Wikipedia page.
+
+### DICOM: Bits Allocated, Bits Stored and High Bit
+
+A question that is often asked on insight-users is: I am trying to
+read/write a DICOM image, but some of the DICOM fields have changed
+after I write it back on disk.
+
+Here is what you have on disk:
+
+```
+Bits Allocated = 16
+Bits Stored    = 12
+High Bit       = 11
+
+                  |<------------------ pixel ----------------->|
+    ______________ ______________ ______________ ______________
+   |XXXXXXXXXXXXXX|              |              |              |
+   |______________|______________|______________|______________|
+    15          12 11           8 7            4 3            0
+```
+
+Here is what you generally get after writing by GDCM:
+
+```
+Bits Allocated = 16
+Bits Stored    = 16
+High Bit       = 15
+
+   |<------------------------- pixel ------------------------->|
+    ______________ ______________ ______________ ______________
```

Comment:
Changed span of `pixel`  dashes in here to cover the whole length. Not sure if the high bit was meant to be included.